### PR TITLE
Update tonic to version 0.12.3 and jsonrpsee to 0.24

### PR DESCRIPTION
**Pull request by sydhds**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

File: massa-sdk/src/lib.rs [hunk 1]
```diff
@@ -642,18 +643,22 @@ impl RpcClientV2 {
 }
 
 fn http_client_from_url(url: &str, http_config: &HttpConfig) -> HttpClient<HttpBackend> {
-    let mut builder = HttpClientBuilder::default()
+    let builder = HttpClientBuilder::default()
         .max_request_size(http_config.client_config.max_request_body_size)
         .request_timeout(http_config.client_config.request_timeout.to_duration())
-        .max_concurrent_requests(http_config.client_config.max_concurrent_requests)
+        // .max_concurrent_requests(http_config.client_config.max_concurrent_requests)
```
Comment (Leo-Besancon) [inline massa-sdk/src/lib.rs:649]: There isn't this config anymore? What's the impact?

**Review by Leo-Besancon — COMMENTED** [review]
I see that proto-rs needed an update too, from what I see it doesn't touch on the serialization of our structs, do you confirm? (that would mean it's not breaking and we don't need to have versioning, otherwise it's very tricky I think)

File: massa-execution-worker/Cargo.toml [hunk 2]
```diff
@@ -89,7 +89,8 @@ tempfile = { workspace = true, optional = true }
 massa_wallet = { workspace = true }
 massa-proto-rs = { workspace = true }
 schnellru = { workspace = true }
-prost = { version = "=0.12", optional = true }
+# prost = { version = "=0.13", optional = true }
```
Comment (bilboquet) [inline massa-execution-worker/Cargo.toml:92]: comment :)

File: massa-metrics/Cargo.toml [hunk 3]
```diff
@@ -9,7 +9,8 @@ sandbox = []
 
 [dependencies]
 prometheus = {workspace = true, "features" = ["process"]}
-hyper = {workspace = true, "features" = ["server", "http1"]}   # BOM UPGRADE     Revert to {"version": "0.14.26", "features": ["server", "tcp", "http1"]} if problem
+# hyper = {workspace = true, "features" = ["server", "http1"]}   # BOM UPGRADE     Revert to {"version": "0.14.26", "features": ["server", "tcp", "http1"]} if problem
```
Comment (bilboquet) [inline massa-metrics/Cargo.toml:12]: comment :)

Comment (sydhds) [reply massa-metrics/Cargo.toml:12]: Should keep this one as we need a quick way to switch between "hyper workspace" and "hyper local version"

File: massa-execution-worker/Cargo.toml (hunk 2 above)
Comment (sydhds) [reply massa-execution-worker/Cargo.toml:92]: Done

File: massa-sdk/src/lib.rs (hunk 1 above)
Comment (sydhds) [reply massa-sdk/src/lib.rs:649]: Unfortunately the field max_concurrent_requests is now private (default is set to: 256). I'll make a PR for jsonrpsee and in the mean time will document this comment.

Comment (Leo-Besancon) [reply massa-sdk/src/lib.rs:650]: Perfect thanks ! I'll keep this comment unresolved to avoid forgetting

File: Cargo.toml [hunk 4]
```diff
@@ -106,8 +106,10 @@ massa_versioning = { path = "./massa-versioning" }
 massa_wallet = { path = "./massa-wallet" }
 
 # Massa projects dependencies
-massa-proto-rs = { git = "https://github.com/massalabs/massa-proto-rs", "rev" = "38950875a7aa406fedc4f0b8336864e5ff290f2c" }
-massa-sc-runtime = { git = "https://github.com/massalabs/massa-sc-runtime", "branch" = "next_breaking_update" }
+massa-proto-rs = { git = "https://github.com/massalabs/massa-proto-rs", "rev" = "48295f3bb23a17075b7be9b4c39f249ef77bef40" }
+# massa-proto-rs = { path = "../massa-proto-rs" }
```
Comment (modship) [inline Cargo.toml:110]: remove

File: Cargo.toml [hunk 5]
```diff
@@ -106,8 +106,10 @@ massa_versioning = { path = "./massa-versioning" }
...
+massa-sc-runtime = { git = "https://github.com/massalabs/massa-sc-runtime", "branch" = "feature/update_tonic_12_3_v2" }
+# massa-sc-runtime = { path = "../massa-sc-runtime_3" }
```
Comment (modship) [inline Cargo.toml:112]: remove

File: massa-sdk/src/lib.rs (hunk 1 above)
Comment (modship) [reply massa-sdk/src/lib.rs:650]: we need to wait your PR to fix this before merge on main

Comment (sydhds) [reply massa-sdk/src/lib.rs:650]: https://github.com/paritytech/jsonrpsee/pull/1470

File: Cargo.toml (hunk 4 above)
Comment (sydhds) [reply Cargo.toml]: Fixed

File: Cargo.toml (hunk 5 above)
Comment (sydhds) [reply Cargo.toml]: Fixed

**Pull request merged**